ddl/check.lua

```diff
@@ -628,75 +628,113 @@ local function check_index(i, index, space)
     return true
 end
 
-local function check_sharding_key(space)
-    if not space.sharding_key then
-        if space.fields.bucket_id ~= nil then
-            return nil, string.format(
-                "spaces[%q].format[%q]: bucket_id is used for sharding, " ..
-                "but there's no spaces[%q].sharding_key defined",
-                space.name, 'bucket_id', space.name
-            )
-        end
-
-        if space.indexes.bucket_id ~= nil then
-            return nil, string.format(
-                "spaces[%q].indexes[%q]: bucket_id is used for sharding, " ..
-                "but there's no spaces[%q].sharding_key defined",
-                space.name, 'bucket_id', space.name
-            )
-        end
-        return true
-    end
-
+local function check_bucket_id_is_valid(space_name, bucket_id_field, bucket_id_index)
```

Comment:
Can we pass just `space` object instead of those three variables? It would make the code a bit easier: `space.fields.bucket_id` and `space.indexes.bucket_id` are more or less obvious from the name, while, say, `bucket_id_field` may be interpreted as a tuple field value or as a tuple field descriptor.

BTW, the idea of the method is to check two different things: correctness of the 'bucket_id' field in a sharded space and correctness of the 'buckte_id' index in a sharded space. They're look strange under one function and the name of the function does not reveal the idea (just because it is hard to do so in a few words). I hope there is away to arrange this code better.

Reply:
I split `check_bucket_id_is_valid` to `check_bucket_id_field_value(space)` and `check_bucket_id_index(space)`